### PR TITLE
Term module implementation

### PR DESCRIPTION
**Pull request by briandowns**
Signed-off-by: Brian Downs <brian.downs@gmail.com>

### Well detailed description of the change :

Adds a new module "Term" with functions to get information about the attached terminal.

Resolves: #

#

### Type of change:

- [ ] Bug fix

- [ ] New feature

- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#

### Housekeeping

- [ ] Tests have been updated to reflect the changes done within this PR (if applicable).

- [ ] Documentation has been updated to reflect the changes done within this PR (if applicable).

#

### CODE REVIEW

File: src/optionals/term.c [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+    ObjString *name = copyString(vm, "Term", 4);
+    push(vm, OBJ_VAL(name));
+    ObjModule *module = newModule(vm, name);
+    push(vm, OBJ_VAL(module));
+
+    /**
+     * Define Term methods
+     */
+    defineNative(vm, &module->values, "isatty", termIsattyNative);
+    defineNative(vm, &module->values, "getSize", getSizeNative);
```
Comment (Jason2605) [inline src/optionals/term.c:92]: Naming convention is always one that trips me up (and something I need to be more strict with myself throughout the code base) but I find it's probably good if we stick with a casing as it'll lead to less confusion

File: docs/docs/standard-lib/term.md [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+### Term.getSize()
+
+Returns the number of rows, columns, horizantal and vertical pixels of the attached terminal.
+
+```cs
+Term.getSize();
+print(Term.getSize());
+// {"rows": 13, "columns": 145, "horizantal_pixels": 145, "vertical_pixels": 145}
+```
```
Comment (Jason2605) [inline docs/docs/standard-lib/term.md:43]:
```suggestion
Returns the number of rows, columns, horizontal and vertical pixels of the attached terminal.

```cs
Term.getSize();
print(Term.getSize());
// {"rows": 13, "columns": 145, "horizontal_pixels": 145, "vertical_pixels": 145}
```
```

File: src/optionals/term.c [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+    int width = (int)(csbi.srWindow.Right-csbi.srWindow.Left+1);
+
+    dictSet(vm, terminalSizeDict, OBJ_VAL(lines), NUMBER_VAL(height));
+    pop(vm);
+
+    dictSet(vm, terminalSizeDict, OBJ_VAL(columns), NUMBER_VAL(width));
+    pop(vm);
+#else
+    ObjString *horizantalPixesl = copyString(vm, "horizantal_pixels", 17);
+    push(vm, OBJ_VAL(horizantalPixesl));
```
Comment (Jason2605) [inline src/optionals/term.c:56]:
```suggestion
    push(vm, OBJ_VAL(horizontalPixesl));
```

File: src/optionals/term.c [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+    struct winsize w;
+    ioctl(0, TIOCGWINSZ, &w);
+
+    dictSet(vm, terminalSizeDict, OBJ_VAL(lines), NUMBER_VAL(w.ws_row));
+    pop(vm);
+
+    dictSet(vm, terminalSizeDict, OBJ_VAL(columns), NUMBER_VAL(w.ws_col));
+    pop(vm);
+
+    dictSet(vm, terminalSizeDict, OBJ_VAL(horizantalPixesl), NUMBER_VAL(w.ws_col));
```
Comment (Jason2605) [inline src/optionals/term.c:70]:
```suggestion
    dictSet(vm, terminalSizeDict, OBJ_VAL(horizontalPixesl), NUMBER_VAL(w.ws_col));
```

File: tests/term/getSize.du [hunk 5]
```diff
@@ -0,0 +1,24 @@
+/**
+ * getSize.du
+ *
+ * Testing the Term.getSize() function
+ *
+ * getSize() returns the number of rows, columns, horizantal and vertical pixels.
```
Comment (Jason2605) [inline tests/term/getSize.du:6]:
```suggestion
 * getSize() returns the number of rows, columns, horizontal and vertical pixels.
```

**Review by Jason2605 — COMMENTED** [review]
Super minor comments again, thanks for this!

File: src/optionals/term.c (hunk 1 above)
Comment (briandowns) [reply src/optionals/term.c:92]: The `isatty` function really threw me for casing. Open to all suggestions.

**Comment by Jason2605** [comment]
The tests are missing from `runTests.du` but I'm going to merge this anyways. I've created a really dirty script that I'll add to `.github` but need something to test with, will create a new PR for it

**Pull request merged**